options/options.go

```diff
@@ -463,6 +463,56 @@ func (opts *ToolOptions) ParseArgs(args []string) ([]string, error) {
 	return args, err
 }
 
+// LogSensitiveOptionWarnings logs a warning for any sensitive information (i.e. passwords)
+// that appear on the command line for the --password, --uri and --sslPEMKeyPassword options.
+// This also applies to a connection string that appears as a positional argument.
+func LogSensitiveOptionWarnings(args []string) {
+	passwordMsg := "WARNING: On some systems, a password provided directly using " +
+		"--password may be visible to system status programs such as `ps` that may be " +
+		"invoked by other users. Consider omitting the password to provide it via stdin, " +
+		"or using the --config option to specify a configuration file with the password."
+
+	uriMsg := "WARNING: On some systems, a password provided directly in a connection string " +
+		"or using --uri may be visible to system status programs such as `ps` that may be " +
+		"invoked by other users. Consider omitting the password to provide it via stdin, " +
+		"or using the --config option to specify a configuration file with the password."
+
+	sslMsg := "WARNING: On some systems, a password provided directly using --sslPEMKeyPassword " +
+		"may be visible to system status programs such as `ps` that may be invoked by other users. " +
+		"Consider using the --config option to specify a configuration file with the password."
+
+	// Create temporary options for parsing command line args.
+	tempOpts := New("", "", "", "", true, EnabledOptions{Auth: true, Connection: true, URI: true})
```

Comment:
> could we add a comment somewhere explaining why we're setting these back to the empty string?

Using the original opts creates some difficulties if you parse into them more than once (the `setURIFromPositionalArg` checks for the connection string already being set), which is why I was resetting them originally. But I agree it's confusing without a comment.

So I'm thinking of just using new opts altogether for the warnings. Should be a little cleaner than resetting each option we issue a warning for.

Reply:
I like this approach!